File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOBpCalControlModel.kt
//
//  IDOBpCalControlModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///

// MARK: - IDOBpCalControlModel

open class IDOBpCalControlModel(
    errorCode: Int,
    operate: Int,
    sbpPpgFeatureNum: Int,
    dbpPpgFeatureNum: Int,
    sbpPpgFeatureItems: List<Int>,
    dbpPpgFeatureItems: List<Int>
) : IDOBaseModel {

    /// Error code: 0 for success, non-zero for failure
    @SerializedName("error_code")
    var errorCode: Int = errorCode

    /// Operation
    /// ```
    /// 0: Invalid
    /// 1: Start blood pressure calibration
    /// 2: Stop blood pressure calibration
    /// 3: Get feature vector
    /// ```
    @SerializedName("operate")
    var operate: Int = operate

    /// Number of high blood pressure PPG feature vectors
    /// Valid when operate=3
    @SerializedName("sbp_ppg_feature_num")
    var sbpPpgFeatureNum: Int = sbpPpgFeatureNum

    /// Number of low blood pressure PPG feature vectors
    /// Valid when operate=3
    @SerializedName("dbp_ppg_feature_num")
    var dbpPpgFeatureNum: Int = dbpPpgFeatureNum

    /// Array of high blood pressure PPG feature vectors
    /// Valid when operate=3
    @SerializedName("sbp_ppg_feature_items")
    var sbpPpgFeatureItems: List<Int> = sbpPpgFeatureItems

    /// Array of low blood pressure PPG feature vectors
    /// Valid when operate=3
    @SerializedName("dbp_ppg_feature_items")
    var dbpPpgFeatureItems: List<Int> = dbpPpgFeatureItems




    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOBpCalControlModel.swift
//
//  IDOBpCalControlModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/19.
//

import Foundation

// MARK: - IDOBpCalControlModel
@objcMembers
public class IDOBpCalControlModel: NSObject, IDOBaseModel {
    /// Error code: 0 for success, non-zero for failure
    public var errorCode: Int
    /// Operation
    /// ```
    /// 0: Invalid
    /// 1: Start blood pressure calibration
    /// 2: Stop blood pressure calibration
    /// 3: Get feature vector
    /// ```
    public var operate: Int
    /// Number of high blood pressure PPG feature vectors
    /// Valid when operate=3
    public var sbpPpgFeatureNum: Int
    /// Number of low blood pressure PPG feature vectors
    /// Valid when operate=3
    public var dbpPpgFeatureNum: Int
    /// Array of high blood pressure PPG feature vectors
    /// Valid when operate=3
    public var sbpPpgFeatureItems: [Int]
    /// Array of low blood pressure PPG feature vectors
    /// Valid when operate=3
    public var dbpPpgFeatureItems: [Int]

    enum CodingKeys: String, CodingKey {
        case errorCode = "error_code"
        case operate = "operate"
        case sbpPpgFeatureNum = "sbp_ppg_feature_num"
        case dbpPpgFeatureNum = "dbp_ppg_feature_num"
        case sbpPpgFeatureItems = "sbp_ppg_feature_items"
        case dbpPpgFeatureItems = "dbp_ppg_feature_items"
    }

    public init(errorCode: Int, operate: Int, sbpPpgFeatureNum: Int, dbpPpgFeatureNum: Int, sbpPpgFeatureItems: [Int], dbpPpgFeatureItems: [Int]) {
        self.errorCode = errorCode
        self.operate = operate
        self.sbpPpgFeatureNum = sbpPpgFeatureNum
        self.dbpPpgFeatureNum = dbpPpgFeatureNum
        self.sbpPpgFeatureItems = sbpPpgFeatureItems
        self.dbpPpgFeatureItems = dbpPpgFeatureItems
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}
